docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -271,31 +205,6 @@ def context_asset(context: AssetExecutionContext):
     ...
 ```
 
-### Conditional materialization
```

Comment:
Will this still be valid if someone does elect to use an I/O manager? Should this be deleted entirely, or just moved elsewhere and re-written here?

Reply:
Ah, yes. This section should get moved to the I/O manager concept page.

I need to do a little testing to ensure i fully understand the syntax for indicating a conditional materialization when not using I/O managers, and then I can write the replacement section

Reply:
hmmm the API to do without I/O managers isn't great....

Reply:
https://github.com/dagster-io/dagster/issues/15418 

Either way, I think the section as written right now should be moved to the I/O manager concept page. And then we can either:
1. be done. Decide that the API ergonomics need to be fixed before publicly documenting this use case for non I/O managed assets
2. write a replacement section for non I/O managed assets that uses the less-than-great APIs

I'll think a bit on what my opinion is for which option we take, but if you have a rule of thumb for this type of situation in the docs, I will follow that!

Reply:
I think we can punt on that so it doesn't hold up this PR.

I don't have a hard and fast rule for handling this kind of thing in the docs. I tend to take into account how often users would ask about something when deciding whether to include a less-than-ideal approach to something in the docs, especially if we're going to change it soon.

Perhaps we could throw the non-ideal workaround in a discussion (which IMO a 'less official' set of content) until the ergonomics are addressed? It could be something like "We plan on improving this thing, but for now, here's how you can do it"

Reply:
I like the discussion approach! i'll merge this and then write that up